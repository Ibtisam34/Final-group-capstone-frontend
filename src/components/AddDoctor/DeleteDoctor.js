import { useEffect, useState } from 'react';
import { FaArrowLeft } from 'react-icons/fa';
import { useSelector, useDispatch } from 'react-redux';
import { Link, useParams, useNavigate } from 'react-router-dom';
import { fetchDoctors } from '../../redux/doctors/doctorsReducer';
import { deleteDoctor } from '../../redux/doctors/doctorsDeleteReducer';

function Removedoctor() {
  const [doctorId, setDoctorId] = useState();
  const { user_id: userId } = useParams();
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleDelete = () => {
    dispatch(deleteDoctor(doctorId));
    dispatch(fetchDoctors());
    navigate(`/${userId}`);
  };

  useEffect(() => {
    dispatch(fetchDoctors());
  }, [dispatch]);

  const doctorList = useSelector((state) => state.doctorsReducer);

  return (

    <div
      className="reservations-container"
      style={{
        backgroundPosition: 'center',
        backgroundSize: 'cover',
        padding: '2rem',
      }}
    >

      <div
        className="reservation-header-links"
        style={{
          position: 'fixed',
          top: '0',
        }}
      >
        <Link to="/"><FaArrowLeft style={{ color: '#41464b', fontSize: '2rem', margin: '5px' }} /></Link>
      </div>

      <form
        style={{
          display: 'flex',
          alignItems: 'center',
          flexDirection: 'column',
        }}
        className="form"
        onSubmit={handleDelete}
      >
        <h1>Remove a doctor from our Listing</h1>

        <select
          style={{
            outline: 'none',
            width: '95%',
            borderRadius: '2px',
            margin: '2rem 0',
          }}
          onChange={(e) => setDoctorId(e.target.value)}
          placeholder="Choose a doctor you want to reserve."
          id="selectDropdown"
        >
          <option disabled selected>
            Choose a doctor
          </option>
          {doctorList != null && doctorList.map((doctor) => (
            <option key={doctor.id} value={doctor.id}>
              {doctor.name}
            </option>
          ))}
        </select>
        <button to="/" onClick={() => { handleDelete(); }} className="submit book-btn" type="submit">Submit</button>
      </form>

    </div>

  );
}

export default Removedoctor;
