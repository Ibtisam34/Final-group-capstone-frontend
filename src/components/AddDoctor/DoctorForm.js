import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Form, Button } from 'react-bootstrap';
import { postDoctor } from '../../redux/doctors/doctorsReducer';
import 'bootstrap/dist/css/bootstrap.min.css';
import '../../App.css';

function CreateDoctorForm() {
  const [name, setName] = useState('');
  const [specialization, setSpecialization] = useState('');
  const [email, setEmail] = useState('');
  const [phone, setPhone] = useState('');
  const [availability, setAvailability] = useState([]);
  const [image, setImage] = useState('');

  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();

    const newDoctor = {
      name,
      specialization,
      email,
      phone,
      availability,
      image,
    };

    dispatch(postDoctor(newDoctor));
  };

  return (
    <Form onSubmit={handleSubmit}>
      <Form.Control
        type="text"
        placeholder="Doctor name"
        className="w-50 mb-3"
        onChange={(e) => setName(e.target.value)}
        required
      />

      <Form.Control
        type="text"
        placeholder="Doctor specialization"
        className="w-50 mb-3"
        onChange={(e) => setSpecialization(e.target.value)}
        required
      />

      <Form.Control
        type="email"
        placeholder="Email"
        className="w-50 mb-3"
        onChange={(e) => setEmail(e.target.value)}
        required
      />

      <Form.Control
        type="tel"
        placeholder="Phone number"
        className="w-50 mb-3"
        onChange={(e) => setPhone(e.target.value)}
        required
      />

      <Form.Control
        type="text"
        placeholder="Availability"
        className="w-50 mb-3"
        onChange={(e) => setAvailability(e.target.value)}
        required
      />

      <Form.Control
        type="text"
        placeholder="Image url"
        className="w-75 mb-3"
        onChange={(e) => setImage(e.target.value)}
        required
      />

      <div className="text-center mb-5">
        <Button className="theme-btn rounded" type="submit">
          Add Doctor
        </Button>
      </div>
    </Form>
  );
}

export default CreateDoctorForm;
