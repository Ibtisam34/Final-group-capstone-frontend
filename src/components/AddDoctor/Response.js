import React from 'react';
import { Alert } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { FaArrowLeft } from 'react-icons/fa';
import PropTypes from 'prop-types';

const Response = ({ show, setShow }) => {
  if (show) {
    return (
      <>
        <div className="doctor-arrow">
          <Link to="/">
            <FaArrowLeft style={{ color: '#41464b', fontSize: '2rem', margin: '5px' }} />
          </Link>
        </div>
        <Alert variant="success" onClose={() => setShow(false)} dismissible>
          <Alert.Heading>Congratulations!!!</Alert.Heading>
          <p>New doctor has been successfully created.</p>
        </Alert>
      </>
    );
  }
  return <add />;
};

Response.propTypes = {
  show: PropTypes.bool.isRequired,
  setShow: PropTypes.func.isRequired,
};

export default Response;
