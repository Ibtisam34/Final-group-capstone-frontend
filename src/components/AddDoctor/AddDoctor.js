import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Container, Button } from 'react-bootstrap';
import CreateDoctorForm from './DoctorForm';
import Response from './Response';
function AddDoctor() {
  const doctors = useSelector((state) => state.doctorsReducer);
  const [show, setShow] = useState(true);
  const handleGoBack = () => {
    window.history.back();
  };
  return (
    <>
      <h1 className="text-center m-4">Add Doctor</h1>
      <Container>
        {doctors.status === 201 ? (
          <Response show={show} setShow={setShow} />
        ) : (
          <>
            <CreateDoctorForm />
            <Button variant="secondary" onClick={handleGoBack}>Go Back</Button>
          </>
        )}
      </Container>
    </>
  );
}
export default AddDoctor;