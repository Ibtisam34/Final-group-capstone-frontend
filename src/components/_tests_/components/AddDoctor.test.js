import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/configureStore';
import AddDoctor from '../../AddDoctor/AddDoctor';

describe('Add a Doctor', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><AddDoctor /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
