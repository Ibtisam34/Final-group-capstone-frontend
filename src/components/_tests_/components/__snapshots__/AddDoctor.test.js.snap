// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add a Doctor renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-center m-4"
      >
        Add Doctor
      </h1>
      <div
        class="container"
      >
        <form
          class=""
        >
          <input
            class="w-50 mb-3 form-control"
            placeholder="Doctor name"
            required=""
            type="text"
          />
          <input
            class="w-50 mb-3 form-control"
            placeholder="Doctor specialization"
            required=""
            type="text"
          />
          <input
            class="w-50 mb-3 form-control"
            placeholder="Email"
            required=""
            type="email"
          />
          <input
            class="w-50 mb-3 form-control"
            placeholder="Phone number"
            required=""
            type="tel"
          />
          <input
            class="w-50 mb-3 form-control"
            placeholder="Availability"
            required=""
            type="text"
          />
          <input
            class="w-75 mb-3 form-control"
            placeholder="Image url"
            required=""
            type="text"
          />
          <div
            class="text-center mb-5"
          >
            <button
              class="theme-btn rounded btn btn-primary"
              type="submit"
            >
              Add Doctor
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text-center m-4"
    >
      Add Doctor
    </h1>
    <div
      class="container"
    >
      <form
        class=""
      >
        <input
          class="w-50 mb-3 form-control"
          placeholder="Doctor name"
          required=""
          type="text"
        />
        <input
          class="w-50 mb-3 form-control"
          placeholder="Doctor specialization"
          required=""
          type="text"
        />
        <input
          class="w-50 mb-3 form-control"
          placeholder="Email"
          required=""
          type="email"
        />
        <input
          class="w-50 mb-3 form-control"
          placeholder="Phone number"
          required=""
          type="tel"
        />
        <input
          class="w-50 mb-3 form-control"
          placeholder="Availability"
          required=""
          type="text"
        />
        <input
          class="w-75 mb-3 form-control"
          placeholder="Image url"
          required=""
          type="text"
        />
        <div
          class="text-center mb-5"
        >
          <button
            class="theme-btn rounded btn btn-primary"
            type="submit"
          >
            Add Doctor
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
